elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryCancelThread.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.service;
+
+import com.yahoo.elide.Elide;
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.QueryStatus;
+import com.yahoo.elide.core.DataStoreTransaction;
+import com.yahoo.elide.core.TransactionRegistry;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Collection;
+import java.util.Date;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Runnable thread for cancelling AsyncQuery transactions
+ * beyond the max run time or if it has status CANCELLED.
+ */
+@Slf4j
+@Data
+@AllArgsConstructor
+public class AsyncQueryCancelThread implements Runnable {
+
+    private int maxRunTimeSeconds;
+    private Elide elide;
+    private AsyncQueryDAO asyncQueryDao;
+
+    @Override
+    public void run() {
+        cancelAsyncQuery();
+    }
+
+    /**
+     * This method cancels queries based on threshold.
+     */
+    protected void cancelAsyncQuery() {
+
+        try {
+            TransactionRegistry transactionRegistry = elide.getTransactionRegistry();
+            Map<UUID, DataStoreTransaction> runningTransactionMap = transactionRegistry.getTransactionMap();
+            Collection<AsyncQuery> asyncQueryCollection = asyncQueryDao.getActiveAsyncQueryCollection();
+
+            for (Map.Entry<UUID, DataStoreTransaction> entry : runningTransactionMap.entrySet()) {
+                for (AsyncQuery obj : asyncQueryCollection) {
+                    if (obj.getRequestId().trim().equals(entry.getKey().toString().trim())) {
+                        Date currentDate = new Date(System.currentTimeMillis());
+                        long diffInMillies = Math.abs(obj.getUpdatedOn().getTime() - currentDate.getTime());
+                        long diffInSecs = TimeUnit.SECONDS.convert(diffInMillies, TimeUnit.MILLISECONDS);
+                        if (obj.getStatus().equals(QueryStatus.CANCELLED)) {
+                            log.debug("Async Query Cancelled: " + obj.getId());
+                            entry.getValue().cancel();
+                        } else if (diffInSecs > maxRunTimeSeconds) {
+                            log.debug("Async Query Cancelled: " + obj.getId());
+                            entry.getValue().cancel();
+                        }
```

Comment:
a couple of things we can probably do to improve the performance here: 
i) When the match occurs in if clause, after processing the if block we should break the inner loop
ii) we should remove the object from asyncQueryCollection when a match happens so that the size keeps reducing as the transaction loop move on.